File: demos/payments/examples/deposit-postback.ts
import express, { Request, Response } from 'express';
import cors from 'cors';

const app = express();
app.use(express.json());

app.use(cors({ origin: '*', allowedHeaders: '*' }));

interface DepositPostbackRequest {
    userId: string;
    amount: number;
    currencyCode: string;
    transactionId: string;
}

app.post('/v1/wallet/deposit-postback', (req: Request, res: Response) => {
    const apiKey = req.header('X-API-KEY');
    const { userId, amount, transactionId }: DepositPostbackRequest = req.body;

    if (!apiKey || apiKey !== 'secretValue') {
        return res.status(401).send('Unauthorized');
    }

    if (amount === undefined || amount === null || amount <= 0) {
        return res.status(400).send('Amount must be positive');
    }
    const newTokens = Math.ceil(amount * 2.5);
    increaseUserBalance(userId, newTokens);
    return res.status(200).send('OK');
});

app.listen(8080, () => console.log('Server running on port 8080'));

File: demos/payments/examples/deposit-postback.py
from fastapi import FastAPI, Header, HTTPException, status
from fastapi.middleware.cors import CORSMiddleware
from pydantic import BaseModel, Field
from decimal import Decimal, ROUND_CEILING
import math

app = FastAPI()

app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

class DepositPostbackRequest(BaseModel):
    userId: str
    amount: Decimal
    currencyCode: str
    transactionId: str

@app.post("/v1/wallet/deposit-postback")
async def handle_deposit_done(
    request: DepositPostbackRequest,
    x_api_key: str = Header(None, alias="X-API-KEY")
):
    if x_api_key != "secretValue":
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Unauthorized"
        )

    if request.amount is None or request.amount <= 0:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail="Amount must be positive"
        )

    raw_tokens = request.amount * Decimal("2.5")
    new_tokens = int(raw_tokens.to_integral_value(rounding=ROUND_CEILING))

    increase_user_balance(request.userId, new_tokens)
    return "OK"
